### ISSUE TITLE
site_url is required, causing a stacktrace

### ISSUE THREAD

**Issue by nicorikken** [iSOLveIT/mkdocs-pdf-generate#15]
I tried adding this plugin in a minimal example, but the code requires a site_url to be set:
https://github.com/iSOLveIT/mkdocs-pdf-generate/blob/ca09a9834deee5d20f59d93214e0fd3e7903ad16/mkdocs_pdf_generate/styles/__init__.py#L52

This now causes a unfriendly stacktrace:
```
ERROR    -  Error building page 'index.md': Error converting index.md to PDF: expected string or bytes-like object
Traceback (most recent call last):
  File "/home/user/.cache/pypoetry/virtualenvs/myrepo-6tCRwMjb-py3.10/lib/python3.10/site-packages/mkdocs_pdf_generate/plugin.py", line 150, in on_post_page
    self.renderer.write_pdf(
  File "/home/user/.cache/pypoetry/virtualenvs/myrepo-6tCRwMjb-py3.10/lib/python3.10/site-packages/mkdocs_pdf_generate/renderer.py", line 37, in write_pdf
    self.render_doc(content, base_url, pdf_metadata=pdf_metadata).write_pdf(
  File "/home/user/.cache/pypoetry/virtualenvs/myrepo-6tCRwMjb-py3.10/lib/python3.10/site-packages/mkdocs_pdf_generate/renderer.py", line 46, in render_doc
    style_tags: list[Tag] = style_for_print(self._options, pdf_metadata)
  File "/home/user/.cache/pypoetry/virtualenvs/myrepo-6tCRwMjb-py3.10/lib/python3.10/site-packages/mkdocs_pdf_generate/styles/__init__.py", line 52, in style_for_print
    _css_escape(re.sub(r"http://|https://", "", options.site_url)),
  File "/usr/lib/python3.10/re.py", line 209, in sub
    return _compile(pattern, flags).sub(repl, string, count)
TypeError: expected string or bytes-like object
```

I'm very new to mkdocs and this pdf-generate plugin, so I don't feel confident to suggest a good fix. I guess a fallback value is needed in case this property isn't set.

I now have set the `site_url` property and it builds fine again.

**Comment by iSOLveIT** [comment]
Hi @nicorikken, 
I will check this out. Thanks for opening the issue.

**Comment by nicorikken** [comment]
How about changing the store in `options.py`:
```python
-        self._site_url = config["site_url"]
+        self._site_url = config.get("site_url", "")
```
But perhaps a warning needs to be triggered, as otherwise all links would fail.

**Comment by iSOLveIT** [comment]
This issue is solved in the commit [bc34a8c](https://github.com/iSOLveIT/mkdocs-pdf-generate/commit/bc34a8c00bc5573f2b1b17f19dc54832b9665b2a).
Thanks for pointing it out.
